### ISSUE TITLE
Choppy scroll on iphone 6

### ISSUE THREAD

**Issue by guarland** [MessageKit/MessageKit#810]
First of all, thanks to the authors of this project. 
 Annoying is that it is very slow scroll, maybe that I'm doing something wrong, I think we can add "Prefetching" ? https://developer.apple.com/documentation/uikit/uicollectionviewdatasourceprefetching/prefetching_collection_view_data

**Comment by SD10** [comment]
@guarland Adopting pre-fetching can cause some problems of it's own 🤔 It's really difficult to to analyze scroll performance from project to project. It's very likely that your performance issues are not related to MessageKit specifically

**Comment by guarland** [comment]
@SD10  I have just compiled MessageKit's example without any modify, but scroll still laggy.
I want to get a smooth scrolling like  Telegram or  Whatsapp have, What do you advise?  Really like your library, and I do not want to go with AsyncDisplayKit

**Comment by SD10** [comment]
@guarland You should use the time profiler in instruments to see what the issue could be

**Comment by binloan** [comment]
I will reopen this bug. I compiled the example app which seems to work nicely, however my custom app  has a very choppy scroll. I ran the profiler, but I'm not getting forward with this issue. Can maybe point me to the right direction? I attached a profiler screenshot below. 

<img width="1277" alt="profiler" src="https://user-images.githubusercontent.com/2640637/45927355-38fa5500-bf32-11e8-940a-c244c698b37b.png">

I implemented all three delegates, however I'm just using the datasource where I return, similar to the example, a new message per section:

```swift
extension ChatViewController : MessagesDataSource{
    func currentSender() -> Sender {
        return Sailor.current.mate.sender
    }
    
    func numberOfSections(in messagesCollectionView: MessagesCollectionView) -> Int {
        return conversation.messages.count
    }
    
    func messageForItem(at indexPath: IndexPath, in messagesCollectionView: MessagesCollectionView) -> MessageType {
        return conversation.messages[indexPath.section]
    }
}
```
It looks like the overhead is produced somewhere inside MessageKit, but I can't properly determine where. Any help would be appreciated. As side note: I'm using a split view controller setup.
